doc/api/icu.md

```diff
@@ -0,0 +1,76 @@
+# icu
+
+    Stability: 1 - Experimental
+
+The `icu` module provides an interface to Unicode and Internationalization
+functionality provided by the ICU4C library.
+
+```js
+const icu = require('icu');
+```
+
+## icu.transcode(buf, from_enc, to_enc)
+
+* `buf` {Buffer} A `Buffer` instance
+* `from_enc` {string} The current encoding
+* `to_enc` {string} The target encoding
+
+Re-encodes the given `Buffer` from one character encoding to another. Returns
+a new `Buffer` instance.
+
+## icu.codePointAt(buf, offset, encoding)
+
+* `buf` {Buffer} A `Buffer` instance
+* `offset` {integer} The byte offset position
+* `encoding` The character encoding of the `Buffer` data. Default = `'utf8'`
+
+Returns the Unicode codepoint located at the given offset in the `Buffer`. Works
+even if the offset falls in the middle of a multibyte UTF-8 or UTF-16 character.
+
+*Note*: While `icu.codePointAt()` is similar to `String.prototype.codePointAt`,
+it differs in that `icu.codePointAt()` will return the correct codepoint
+regardless of whether the `offset` points to the start, middle or end of a
+UTF-8 or UTF-16 encoded character.
+
+## icu.charAt(buf, offset, encoding)
+
+* `buf` {Buffer} A `Buffer` instance
+* `offset` {integer} The byte offset position
+* `encoding` The character encoding of the `Buffer` data. Default = `'utf8'`
+
+Returns the character located at the given offset in the `Buffer`. Works even
+if the offset falls in the middle of a multibyte UTF-8 or UTF-16 character.
+
+*Note*: While `icu.charAt()` is similar to `String.prototype.charAt`,
+it differs in that `icu.charAt()` will return the correct character
+regardless of whether the `offset` points to the start, middle or end of a
+UTF-8 or UTF-16 encoded character.
+
+## icu.slice(buf, encoding, start, end)
+
+* `buf` {Buffer} A `Buffer` instance
+* `encoding` {String} A character encoding
+* `start` {integer} The starting character offset
+* `end` {integer} The ending character offset
+
+Performs an encoding aware slice of the Buffer instance. The `start` and `end`
+arguments define character offsets rather than byte offsets. Ensures that the
+slice occurs at proper codepoint boundaries.
+
+## icu.utf8Slice(buf, start, end)
```

Comment:
`icu.slice(buf, 'utf8')` defers to `icu.utf8Slice()`... so there would be no difference. I'll likely be pulling `icu.utf8Slice()` off the exported API.